django/contrib/auth/tokens.py

```diff
@@ -47,14 +72,21 @@ def check_token(self, user, token):
             return False
 
         # Check that the timestamp/uid has not been tampered with
-        if not constant_time_compare(self._make_token_with_timestamp(user, ts), token):
+        attempts = [
+            constant_time_compare(self._make_token_with_timestamp(user, ts, secret), token)
+            for secret in self._get_secrets()
+        ]
+        if not constant_time_any(attempts):
```

Comment:
Is this really necessary? Which information do we leak by using a normal any? To the best of my knowledge you'd leak which `SECRET_KEY` you used, which imo shouldn't be a problem? I also think it is maybe not the best idea to try against all secret keys if the first one (the common one) already matches, this creates extra work that should not be needed imo.

In that sense I'd loop and break out early if you get a valid sig.

Reply:
This was not fully clear when I wrote the original patch either. 🙂 I added constant_time_any just for good measure. I discussed it with @MarkusH at DjangoCon EU in Copenhagen and (IIRC) we figured that it would be a good idea to keep the constant time check just to be extra careful.

That said, yes, it could leak which secret key that was used but... I cannot see why that would be a problem. I do not have a strong opinion either way.